trakt/sync.py

```diff
@@ -65,57 +68,100 @@ def add_to_history(media, watched_at=None):
 
     data = dict(watched_at=timestamp(watched_at))
     data.update(media.ids)
-    yield 'sync/history', {media.media_type: [data]}
+    result = yield 'sync/history', {media.media_type: [data]}
+    yield result
 
 
 @post
 def add_to_watchlist(media):
-    """Add a :class:`TVShow` to your watchlist
-
-    :param media: The :class:`TVShow` object to add to your watchlist
+    """Add a :class:`Movie`, :class:`TVShow`, or :class:`TVEpisode`
+        to your watchlist
+    :param media: Supports both the PyTrakt :class:`Movie`,
+        :class:`TVShow`, etc. But also supports passing custom json structures.
     """
-    yield 'sync/watchlist', media.to_json()
+    from trakt.tv import TVEpisode, TVSeason, TVShow
+    from trakt.movies import Movie
+    if isinstance(media, (TVEpisode, TVSeason, TVShow, Movie)):
+        media_object = media.to_json()
+    else:
+        media_object = media
+
+    result = yield 'sync/watchlist', media_object
+    yield result
 
 
 @post
 def remove_from_history(media):
     """Remove the specified media object from your history
 
-    :param media: The media object to remove from your history
+    :param media: Supports both the PyTrakt :class:`Movie`,
+        :class:`TVShow`, etc. But also supports passing custom json structures.
     """
-    yield 'sync/history/remove', media.to_json()
+    from trakt.tv import TVEpisode, TVSeason, TVShow
+    from trakt.movies import Movie
+    if isinstance(media, (TVEpisode, TVSeason, TVShow, Movie)):
+        media_object = media.to_json()
+    else:
+        media_object = media
 
+    result = yield 'sync/history/remove', media_object
+    yield result
 
-@post
-def remove_from_watchlist(media):
-    """Remove a :class:`TVShow` from your watchlist
 
-    :param media: The :class:`TVShow` to remove from your watchlist
+@post
+def remove_from_watchlist(media=None):
```

Comment:
is `None` actually a valid default value here? I don't see anything in the trakt API docs about supporting null request bodies for this endpoint, which I think is what would happen if someone were to call `remove_from_watchlist()` with no argument.